### ISSUE TITLE
problems about WanVAE reconstruction quality

### ISSUE THREAD

**Issue by leckiepeng** [modelscope/DiffSynth-Studio#498]
I was trying to finetuing the Wan2.1 following train_wan_t2v.py, when i test the encoded latents with the vae.decode, the resulting video doesn't seem right. Bellow is the test code which just decode the latents right after encoding and then save the video.  I think the reconstruction quality would be very high since the latents directly come from the encoder. Any suggestion or answer is appreciated.

@@ -136,6 +136,11 @@ class LightningModelForDataProcess(pl.LightningModule):
             # video
             video = video.to(dtype=self.pipe.torch_dtype, device=self.pipe.device)
             latents = self.pipe.encode_video(video, **self.tiler_kwargs)[0]
\+          recon_video = self.pipe.decode_video(latents, **self.tiler_kwargs)[0]
\+          recon_video = self.pipe.tensor2video(recon_video)
\+          save_video(recon_video, f"recon_video_{np.random.randint(100000)}.mp4", fps=25)
\+          save_frames(recon_video, "recon_frames")
             # image
             if "first_frame" in batch:
                 first_frame = Image.fromarray(batch["first_frame"][0].cpu().numpy())

CUDA_VISIBLE_DEVICES="0" python examples/wanvideo/train_wan_t2v.py \
  --task data_process \
  --dataset_path data/example_dataset \
  --output_path ./models \
  --text_encoder_path "../../Wan2.1-I2V-14B-480P/models_t5_umt5-xxl-enc-bf16.pth" \
  --vae_path "../../Wan2.1-I2V-14B-480P/Wan2.1_VAE.pth" \
  --num_frames 81 \
  --height 832 \
  --width 480

(note that tiling is disabled)

Bellow is one of the reconstructed frames, the original input is a real scene video.
![Image](https://github.com/user-attachments/assets/007af936-325e-46cc-85db-88c178d539e1)

**Comment by Artiprocher** [comment]
@leckiepeng Can you provide your original video files here?

**Comment by leckiepeng** [comment]
> [@leckiepeng](https://github.com/leckiepeng) Can you provide your original video files here?
Thanks for the reply. Here's the test video, I've tried other videos, same problem.

**Comment by leckiepeng** [comment]
> [@leckiepeng](https://github.com/leckiepeng) Can you provide your original video files here?

https://drive.google.com/file/d/1QdCF6Hot3s8ktKzqsUjafPnhmOt7whHk/view?usp=sharing

**Comment by leckiepeng** [comment]
Hi guys, I've also tried the official Wan2.1 github code, the results are different but still way below the par. I was pretty sure about my test code, something roughly like "latents = wan_t2v.vae.model.encode(frames, scale); video = wan_t2v.vae.decode(latents)". Since the original training code is not officially available, we are not sure how images/videos were preprocessed, any mismatch could cause the above problem, for example maybe videos were [-1, 1] normalized when training the VAE, instead of [0, 1] normalized like in this repo.

I also found there are some LoRa finetuned checkponts in the comunity which seem to generate fine, is it because the finetuning process were gentle and did't disrupt the strong backbone?

**Comment by leckiepeng** [comment]
> [@leckiepeng](https://github.com/leckiepeng) Here is my code. We do not notice any significant drop in video quality.
> 
> import torch
> from diffsynth import ModelManager, WanVideoPipeline, save_video, VideoData
> from modelscope import snapshot_download
> 
> 
> # Download models
> snapshot_download("Wan-AI/Wan2.1-T2V-1.3B", local_dir="models/Wan-AI/Wan2.1-T2V-1.3B")
> 
> # Load models
> model_manager = ModelManager(
>     device="cuda", torch_dtype=torch.bfloat16,
>     file_path_list=["models/Wan-AI/Wan2.1-T2V-1.3B/Wan2.1_VAE.pth"],
> )
> pipe = WanVideoPipeline.from_model_manager(model_manager)
> 
> video = VideoData("test.mp4", height=832, width=480)
> video = [video[i] for i in range(81)]
> 
> # Encode
> input_video = pipe.preprocess_images(video)
> input_video = torch.stack(input_video, dim=2).to(dtype=pipe.torch_dtype, device=pipe.device)
> latents = pipe.encode_video(input_video).to(dtype=pipe.torch_dtype, device=pipe.device)
> 
> # Decode
> output_video = pipe.decode_video(latents)
> output_video = pipe.tensor2video(output_video[0])
> save_video(output_video, "test_2.mp4", fps=15, quality=5)


Thanks for the help, my test code was indeed incorrect. pipe.decode_video expects latents of shape [B, T, C, H, W], while the latents with which i called pipe.decode has been squeezed by the [0] indexing. Sorry for the confusion.
